salt/states/boto_ec2.py

```diff
@@ -1752,5 +1749,169 @@ def private_ips_absent(name, network_interface_name=None, network_interface_id=N
 
         # there were no changes since we did not attempt to remove ips
         ret['changes'] = {}
+    return ret
+
+
+def secgroup_present(
```

Comment:
Yeah, that's totally reasonable, and why we have other helper functions like this, but it's also best to have the state properly manage its subresources as well. So, if it's not difficult, it would be ideal for changes to the secgroups in the instance_present function also idempotently maintain the instances secgroups.